### PR TITLE
CXX-2046 Implement operator[] for bsoncxx::document::value

### PR DESCRIPTION
**Pull request by rayangler**

### CODE REVIEW

**Review by bazile-clyde — APPROVED** [review]
LGTM! Great work!

File: src/bsoncxx/test/bson_get_values.cpp [hunk 1]
```diff
@@ -340,4 +340,22 @@ TEST_CASE("CXX-1880: array element should have key") {
...
+    //     "test_array": [5, 4, 3]
+    // }
+    auto doc = make_document(kvp("beep", 25),
+                             kvp("boop", make_document(kvp("test", true))),
+                             kvp("test_array", make_array(5, 4, 3)));
+    auto view = doc.view();
+
+    REQUIRE(doc["beep"].get_int32() == view["beep"].get_int32());
+    REQUIRE(doc["boop"]["test"].get_bool() == view["boop"]["test"].get_bool());
+    REQUIRE(doc["test_array"][2].get_int32() == view["test_array"][2].get_int32());
```
Comment (samantharitter) [inline src/bsoncxx/test/bson_get_values.cpp:358]: Could you please add a test that attempts to find a key that does not exist in the document?

**Review by samantharitter — CHANGES_REQUESTED** [review]
Code changes look fine to me, I just requested an additional test case.

**Review by samantharitter — APPROVED** [review]
LGTM!

**Review by kevinAlbs — APPROVED** [review]
LGTM! This is a good convenience.

**Pull request merged**